src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java

```diff
@@ -18,7 +18,31 @@
 
 package com.mongodb.kafka.connect.sink;
 
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.CHANGE_DATA_CAPTURE_HANDLER_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.CHANGE_DATA_CAPTURE_HANDLER_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DELETE_ON_NULL_VALUES_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DELETE_ON_NULL_VALUES_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_OVERWRITE_EXISTING_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_OVERWRITE_EXISTING_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_UUID_FORMAT_CONFIG;
```

Comment:
I think the topic configuration should be responsible for logging all topic based configuration.  The top level sink configuration should delegate to each topic configuration.

Reply:
The benefits are the topic based configuration manages its own validation and not the higher level sink config.  It seems to be recoupling the design by pushing this upwards.

Reply:
> the topic based configuration manages its own validation

I don't think this is the case: the validation is currently done (even before this PR) by the static code. I.e., there is no object that represents topic-specific configuration that has an instance method validating it. Instead, there is a static method `MongoSinkTopicConfig.validateAll` doing this, and this static method is called from `ConfigDef.validateAll` defined in `MongoSinkConfig`.

In other words, the proposal will result only in restructuring the static code. No responsibility will be moved from from an object to a different object, because validation is done by static code, not instance methods. I.e, it is simply impossible to say that the topic based configuration will manage its own validation, because the object representing that configuration is created after the static code does the validation.

Reply:
We discussed this over Zoom, and agreed to move the logic into a separate class. It was moved to a new `SinkConfigSoftValidator` class, what exposes methods `logObsoleteProperties` and `logIncompatibleProperties`, used from `ConfigDef.validateAll`.